### PR TITLE
feat(cli): binary support for blob type

### PR DESCRIPTION
**Pull request by aaqilniz**
Currently, discovering models that have a field with BLOB type fails the build with an error. 

```
npm run build
> binaryissue@0.0.1 build
> lb-tsc
src/models/amounts.model.ts:69:11 - error TS2304: Cannot find name 'Binary'.
69   image?: Binary;
             ~~~~~~
src/models/amounts.model.ts:69:11 - error TS4031: Public property 'image' of exported class has or is using private name 'Binary'.
69   image?: Binary;
             ~~~~~~
Found 2 errors in the same file, starting at: src/models/amounts.model.ts:69
```

This happens due to the fact that lb4 doesn't handle Binary type. This PR intends to fix this issue and provides support for the BLOB type.

There was [PR](https://github.com/loopbackio/loopback-next/pull/6476) a while ago which was closed without merge. This PR continues that PR and has added a test for the change.

Related GitHub issues & PRs:

Fixes # [6295](https://github.com/loopbackio/loopback-next/issues/6295)
Fixes # [9043](https://github.com/loopbackio/loopback-next/issues/9043)
Fixes # [144](https://github.com/loopbackio/loopback-connector-oracle/issues/144)

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `examples/*` were updated

### CODE REVIEW

File: packages/cli/generators/datasource/index.js
Comment (achrinza) [inline packages/cli/generators/datasource/index.js:68]: Is this necessary for this PR to work?

Otherwise, I think this should be kept as a separate discussion/PR.

File: packages/cli/lib/base-generator.js
Comment (achrinza) [inline packages/cli/lib/base-generator.js:199]: Ditto

File: packages/cli/generators/datasource/index.js
Comment (aaqilniz) [reply packages/cli/generators/datasource/index.js:68]: This is actually coming from the master branch of my fork. I made another PR with these changes.

File: packages/cli/lib/base-generator.js
Comment (aaqilniz) [reply packages/cli/lib/base-generator.js:199]: What happens when this.options[o] is actually an empty string? I've created a repo to reproduce the issue [here](https://github.com/aaqilniz/lb4-datasource-bug).
btw this change is also coming from the master branch. I created another PR with the master branch [here](https://github.com/loopbackio/loopback-next/pull/9014). I think I need to re-create the PRs by creating separate branches for each change.

Comment (aaqilniz) [reply packages/cli/lib/base-generator.js:199]: @achrinza Can you please confirm and guide me if I should be creating the PRs again and closing the existing ones? Then have a separate branch for every change. 

I have got a few more changes that I will be creating PRs for - so how should I proceed about this?

**Comment by aaqilniz** [comment]
@achrinza Can you please confirm and guide me if I should be creating the PRs again and closing the existing ones? Then have a separate branch for every change. 

I have got a few more changes that I will be creating PRs for - so how should I proceed about this?

**Comment by aaqilniz** [comment]
This PR only adds support for Binary support only in models. Hence it doesn't completely fix the problem of supporting BLOB data in LoopBack 4 but it does fix the issues mentioned in the PR.

**Comment by aaqilniz** [comment]
@achrinza Can you please have a look at the PR again?

**Comment by aaqilniz** [comment]
Though this PR doesn't provide complete support for binary type it does at the model level. So can we get this PR merged?

**Review by dhmlau — APPROVED** [review]
@aaqilniz, thanks for the PR. Your changes LGTM. I'd like to get at least one more review before merging. Thanks.

**Review by samarpan-b — APPROVED** [review]

**Comment by dhmlau** [comment]
@achrinza, since you've reviewed this PR before, are you good with merging this PR?

**Comment by aaqilniz** [comment]
Hi @achrinza. Can we get this PR merged if everything looks good?

**Review by achrinza — APPROVED** [review]

**Comment by aaqilniz** [comment]
@dhmlau , It's approved by @achrinza, so can we get this merged?

**Pull request merged**
